scripts/o3de/o3de/repo.py

```diff
@@ -106,27 +109,65 @@ def validate_remote_repo(repo_uri: str, validate_contained_objects: bool = False
             except json.JSONDecodeError as e:
                 logger.error(f'Invalid JSON - {cache_file} could not be loaded')
                 return False
+                          
+        repo_schema_version = get_repo_schema_version(repo_data)
 
-        if download_object_manifests(repo_data) != 0:
-            return False
-
-        gem_set = get_gem_json_paths_from_cached_repo(repo_uri)
-        for gem_json in gem_set:
-            if not validation.valid_o3de_gem_json(gem_json):
-                logger.error(f'Invalid gem JSON - {gem_json} could not be loaded or is missing required values')
-                return False
-
-        project_set = get_project_json_paths_from_cached_repo(repo_uri)
-        for project_json in project_set:
-            if not validation.valid_o3de_project_json(project_json):
-                logger.error(f'Invalid project JSON - {project_json} could not be loaded or is missing required values')
-                return False
-
-        template_set = get_template_json_paths_from_cached_repo(repo_uri)
-        for template_json in template_set:
-            if not validation.valid_o3de_template_json(template_json):
-                logger.error(f'Invalid template JSON - {template_json} could not be loaded or is missing required values')
+        if repo_schema_version == "1.0.0":
+            if download_object_manifests(repo_data) != 0:
                 return False
+            gem_set = get_gem_json_paths_from_cached_repo(repo_uri)
+            for gem_json in gem_set:
+                if not validation.valid_o3de_gem_json(gem_json):
+                    logger.error(f'Invalid gem JSON - {gem_json} could not be loaded or is missing required values')
+                    return False
+            project_set = get_project_json_paths_from_cached_repo(repo_uri)
+            for project_json in project_set:
+                if not validation.valid_o3de_project_json(project_json):
+                    logger.error(f'Invalid project JSON - {project_json} could not be loaded or is missing required values')
+                    return False
+            template_set = get_template_json_paths_from_cached_repo(repo_uri)
+            for template_json in template_set:
+                if not validation.valid_o3de_template_json(template_json):
+                    logger.error(f'Invalid template JSON - {template_json} could not be loaded or is missing required values')
+                    return False
+                
+        elif repo_schema_version == "2.0.0":
```

Comment:
What is the reason for adding a version 2, if version 1 hasn't been added yet?

Reply:
this is related to your other feedback, but basically if we are considering the current repo.json to be version 0.0.0, then you're right, this would have to be 1.0.0 instead. I thought we would be defining the current repo.json to be version 1.0.0, but I can change it around.